agent/data/client.go

```diff
@@ -91,14 +92,16 @@ type Client interface {
 	// GetMetadata gets the value of a certain kind of metadata.
 	GetMetadata(string) (string, error)
 
+	// NetworkDataClient contains networking related data operations
+	generaldata.NetworkDataClient
```

Comment:
Uhh yeah you are right. My original thought was as we will still need the concrete type embed to avoid multiple boltDB or dataclient instance, whether or not avoid the interface embedding here is not important. But from a testing/mocking perspective avoid embedding does make sense to me.